module/settings.mjs

```diff
@@ -348,16 +351,24 @@ export function registerDeferredSettings() {
  * Set the theme on an element, removing the previous theme class in the process.
  * @param {HTMLElement} element  Body or sheet element on which to set the theme data.
  * @param {string} [theme=""]    Theme key to set.
+ * @param {string[]} [flags=[]]  Additional theming flags to set.
  */
-export function setTheme(element, theme="") {
+export function setTheme(element, theme="", flags=new Set()) {
+  element.className = element.className.replace(/\bdnd5e-(theme|flag)-[\w-]+\b/g, "");
+
+  // Primary Theme
   if ( !theme && (element === document.body) ) {
     if ( matchMedia("(prefers-color-scheme: dark)").matches ) theme = "dark";
     if ( matchMedia("(prefers-color-scheme: light)").matches ) theme = "light";
   }
-  element.className = element.className.replace(/\bdnd5e-theme-\w+/g, "");
   if ( theme ) {
     element.classList.add(`dnd5e-theme-${theme.slugify()}`);
     element.dataset.theme = theme;
   }
   else delete element.dataset.theme;
+
+  // Additional Flags
+  if ( matchMedia("(prefers-contrast: more)").matches ) flags.add("high-contrast");
```

Comment:
Noticed this accidentally just now but I think this also needs to go into the `element === document.body` block, otherwise sheets that toggle their themes back to 'Automatic' get stuck with this flag on.